### DISCUSSION TITLE
Browser instances spawned from osu will prevent it from updating

### DISCUSSION THREAD

**Discussion by minhperry** [ppy/osu#30745] [Q&A]
### Type

Other

### Bug description

This happened yesterday and I only remeber about this one day later.

When you spawn a browser instance (for example, Firefox is used in my case) from osu listing or player view, that instance would be "attached" to osu and thus when I tried to update by clicking on the ingame updater, it will fail and say that those Firefox processes are prevented osu from updating.

I think web browser instances should not be attached to the game at all.

### Version

2024.1009.1 -> 2024.1115.2

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17787809/compressed-logs.zip)

although log is not so helpful in this case

**Comment by bdach** [comment]
> that instance would be "attached" to osu and thus when I tried to update by clicking on the ingame updater, it will fail and say that those Firefox processes are prevented osu from updating.

Where are you getting this from? I don't see anything of the sort in logs...?

**Reply by minhperry** [reply]
The log was from another session. I couldn't get the true log since it happened the previous day.

**Reply by minhperry** [reply]
if i can somehow "unupdate" this ill try to replicate it

**Comment by minhperry** [comment]
[compressed-logs.zip](https://github.com/user-attachments/files/18064419/compressed-logs.zip)

idk if these will contain all the logs but here are the steps to reproduce:
1. close all current Firefox browser instances
2. let osu download the new update but dont install it
3. open beatmap listing, select a map and open in browser
4. try to install update

The logs are from exporting immediately after a successful update by closing the browser and let osu update.
@bdach

**Comment by minhperry** [comment]
Another from recent update, done the same way as above
[compressed-logs.zip](https://github.com/user-attachments/files/18311580/compressed-logs.zip)

**Reply by peppy** [reply]
This is specific to you and we haven't heard it reported by anyone else. I think you may need to check your setup. There's no way we can "attach" a browser to the game. It's just not how processes work.

**Reply by minhperry** [reply]
Is there be any way to downgrade the game for further testing?

**Reply by peppy** [reply]
you canjust download an old version from github
